In [1]:
import scipy.stats as stats
import numpy as np
import seaborn as sns
import pandas as pd
import matplotlib.pyplot as plt

First we imported different libraries

In [2]:
data=pd.read_csv('Sales_add.csv')

In [3]:
data

Month      Region      Manager  Sales_before_digital_add(in $)  \
0    Month-1  Region - A  Manager - A                          132921   
1    Month-2  Region - A  Manager - C                          149559   
2    Month-3  Region - B  Manager - A                          146278   
3    Month-4  Region - B  Manager - B                          152167   
4    Month-5  Region - C  Manager - B                          159525   
5    Month-6  Region - A  Manager - B                          137163   
6    Month-7  Region - C  Manager - C                          130625   
7    Month-8  Region - A  Manager - A                          131140   
8    Month-9  Region - B  Manager - C                          171259   
9   Month-10  Region - C  Manager - B                          141956   
10  Month-11  Region - B  Manager - C                          159339   
11  Month-12  Region - A  Manager - A                          178939   
12  Month-13  Region - B  Manager - A                          145062   
13  Month-14  Region - A  Manager - A                          151514   
14  Month-15  Region - C  Manager - A                          147463   
15  Month-16  Region - A  Manager - B                          177195   
16  Month-17  Region - A  Manager - C                          140862   
17  Month-18  Region - C  Manager - B                          167996   
18  Month-19  Region - B  Manager - A                          132135   
19  Month-20  Region - A  Manager - B                          152493   
20  Month-21  Region - B  Manager - A                          147425   
21  Month-22  Region - A  Manager - C                          130263   

    Sales_After_digital_add(in $)  
0                          270390  
1                          223334  
2                          244243  
3                          231808  
4                          258402  
5                          256948  
6                          222106  
7                          230637  
8                          226261  
9                          193735  
10                         203175  
11                         276279  
12                         205180  
13                         253131  
14                         229336  
15                         187305  
16                         234908  
17                         191517  
18                         227040  
19                         212579  
20                         263388  
21                         243020

In [4]:
data.describe()

Sales_before_digital_add(in $)  Sales_After_digital_add(in $)
count                       22.000000                      22.000000
mean                    149239.954545                  231123.727273
std                      14844.042921                   25556.777061
min                     130263.000000                  187305.000000
25%                     138087.750000                  214960.750000
50%                     147444.000000                  229986.500000
75%                     157627.500000                  250909.000000
max                     178939.000000                  276279.000000

# The company wishes to clarify whether there is any increase in sales after stepping into digital marketing.

In [5]:
x=np.array(data['Sales_After_digital_add(in $)'])

In [6]:
y=np.array(data['Sales_before_digital_add(in $)'])

we made an array of x and y to check ttest.we make null hypothesis as no increase in sales and alternative as increase in sales

In [7]:
stats.ttest_ind(x,y,alternative='greater')

Ttest_indResult(statistic=12.995084451110877, pvalue=1.3071840034523225e-16)

pvalue=1.3071840034523225e-16 Hence we reject our null hypothesis. and there is increase in sales after stepping into digital marketing

# The company needs to check whether there is any dependency between the features “Region” and “Manager”.

Here null hypothesis:  there is no dependency between the features “Region” and “Manager”.
Alternative Hypothesis:  there is dependency between the features “Region” and “Manager”.

In [8]:
data['Region'].value_counts()

Region - A    10
Region - B     7
Region - C     5
Name: Region, dtype: int64

In [9]:
data['Manager'].value_counts()

Manager - A    9
Manager - B    7
Manager - C    6
Name: Manager, dtype: int64

In order to compute the Chi-square test statistic, we would need to construct a contingency table.

In [16]:
pd.crosstab(data.Region, data.Manager)

Manager     Manager - A  Manager - B  Manager - C
Region                                           
Region - A            4            3            3
Region - B            4            1            2
Region - C            1            3            1

The numbers in this table represent frequencies.

In [25]:
ct = pd.crosstab(data.Region, data.Manager)
stat,p,dof,expected=stats.chi2_contingency(ct)
print(f'chisquare value:{stat}')
print(f'p value:{p}')
print(f'degree of freedom:{dof}')
print(f'expected values:{expected}')

chisquare value:3.050566893424036
p value:0.5493991051158094
degree of freedom:4
expected values:[[4.09090909 3.18181818 2.72727273]
 [2.86363636 2.22727273 1.90909091]
 [2.04545455 1.59090909 1.36363636]]


we got p value as .549 which is greater than .05 so weaccept null hypothesis.therefore both are independent